src/material/button/button-ripple.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ANIMATION_MODULE_TYPE,
+  Directive,
+  ElementRef,
+  Inject,
+  NgZone,
+  Optional,
+  inject,
+} from '@angular/core';
+import {
+  MAT_RIPPLE_GLOBAL_OPTIONS,
+  MatRipple,
+  RippleGlobalOptions,
+  RippleRenderer,
+  RippleTarget,
+} from '@angular/material/core';
+import {DOCUMENT} from '@angular/common';
+import {Platform} from '@angular/cdk/platform';
+import {MatButtonLazyLoader} from './button-lazy-loader';
+
+/**
+ * The MatButtonRipple directive is an extention of the MatRipple
+ * which allows us to lazily append the DOM node for the button ripple.
+ *
+ * The MatButtonRipple directive allows us to not immediately attach the ripple node to the button by providing a separate render function
+ */
+@Directive({
+  selector: '[mat-button-ripple], [matButtonRipple]',
+  exportAs: 'matButtonRipple',
+  standalone: true,
+})
+export class MatButtonRipple extends MatRipple {
+  /** The host button element. */
+  private _buttonEl: HTMLElement;
+
+  /** The element that the MatRipple is attached to. */
+  private _rippleEl: HTMLElement;
+
+  /** Whether this ripple has already been attached. */
+  _isAttached = false;
+
+  _rippleLoader = inject(MatButtonLazyLoader);
+
+  constructor(
+    _elementRef: ElementRef<HTMLElement>,
+    readonly ngZone: NgZone,
+    readonly platform: Platform,
+    @Optional() @Inject(MAT_RIPPLE_GLOBAL_OPTIONS) globalOptions?: RippleGlobalOptions,
+    @Optional() @Inject(ANIMATION_MODULE_TYPE) _animationMode?: string,
+    @Optional() @Inject(DOCUMENT) document?: Document,
+  ) {
+    super(_elementRef, ngZone, platform, globalOptions, _animationMode, document);
+    this._buttonEl = _elementRef.nativeElement;
+    this._buttonEl.classList.remove('mat-ripple');
+  }
```

Comment:
Prefer the `inject` function

Reply:
I was able to do this for `document`, but not any of the other symbols because they are referenced in the call to `super` and references to `this` are not allowed before `super` is called